drivers/scheduler/scheduler.go

```diff
@@ -189,6 +189,21 @@ type Driver interface {
 
 	// GetSpecAppEnvVar gets app environment variable value by given key name
 	GetSpecAppEnvVar(ctx *Context, key string) string
+
+	// CreateAutopilotRule creates the AutopilotRule object
+	CreateAutopilotRule(apRule apapi.AutopilotRule) (*apapi.AutopilotRule, error)
+
+	// GetAutopilotRule gets the AutopilotRule for the provided name
+	GetAutopilotRule(name string) (*apapi.AutopilotRule, error)
+
+	// UpdateAutopilotRule updates the AutopilotRule
+	UpdateAutopilotRule(apapi.AutopilotRule) (*apapi.AutopilotRule, error)
+
+	// DeleteAutopilotRule deletes the AutopilotRule of the given name
+	DeleteAutopilotRule(name string) error
+
+	// ListAutopilotRules lists AutopilotRules
+	ListAutopilotRules() (*apapi.AutopilotRuleList, error)
```

Comment:
it makes sense to add these api functions only if you want to give tests control on autopilot rules, otherwise if you're not calling these functions from outside the K8s driver, you don't need to modify api at this point (you can always add them later)